sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ConvertHelpers.java

```diff
@@ -57,9 +61,20 @@
   "rawtypes"
 })
 public class ConvertHelpers {
-  private static class SchemaInformationProviders {
-    private static final ServiceLoader<SchemaInformationProvider> INSTANCE =
-        ServiceLoader.load(SchemaInformationProvider.class);
+
+  private static final AtomicReference<List<SchemaInformationProvider>>
+      SCHEMA_INFORMATION_PROVIDERS = new AtomicReference<>();
+
+  private static List<SchemaInformationProvider> getSchemaInformationProviders() {
+    return SCHEMA_INFORMATION_PROVIDERS.updateAndGet(
```

Comment:
this makes List<SchemaInformationProvider> only update once. I'm not sure if it is possible if iterating over "SchemaInformationProviders.INSTANCE" would return different values overtime currently

forwarding to experts who I remember reviewed / fixed concurrency issues  R: @scwhittle @kennknowles thanks!

Reply:
I'm also not sure when providers will be registered.  From ServiceLoader docs it caches internally but looks for new things to load each time. 

It seems like a smaller change to semantics to synchronize access to the loader.
Can you add a static Object to use as a lock, annotate INSTANCE as GuardedBy it and  synchronize iteration?

Reply:
Done